### PR TITLE
perf: use hashing to determine the format in infer/lib/audio.py

### PR DESCRIPTION
**Pull request by 413x1nkp**
this makes use of python's dictionaries, comprehension of which is much more performant than the usage of if-elif-else 
- added type-safety for the functions
- added additional check on whether or not the file exists to not attempt to create a pipe if it doesn't (mitigates expensive subprocess call)

### CODE REVIEW

**Comment by 413x1nkp** [comment]
Please squash in case of PR being accepted! I'm using Python3.12 as a primary python version, some things like type subscription simply don't work in some of the tests intended for Python<3.9, hence the multiple `fix` commits

File: infer/lib/audio.py [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
+audio_format_dict: Dict[str, str] = {
+    "ogg": "libvorbis",
+    "mp4": "aac",
+}
+
+def wav2(i: BytesIO, o: BufferedWriter, format: str):
     inp = av.open(i, "r")
-    if format == "m4a":
-        format = "mp4"
+    format = video_format_dict[format]
```
Comment (fumiama) [inline infer/lib/audio.py]: I think the dict will panic if the format is not m4a. You can use the `get` grammar and set the default value as `format` itself.

File: infer/lib/audio.py [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
     out = av.open(o, "w", format=format)
-    if format == "ogg":
-        format = "libvorbis"
-    if format == "mp4":
-        format = "aac"
+    format = audio_format_dict[format]
```
Comment (fumiama) [inline infer/lib/audio.py]: Same issue.

**Review by fumiama — CHANGES_REQUESTED** [review]

File: infer/lib/audio.py (hunk 1 above)
Comment (413x1nkp) [reply infer/lib/audio.py]: Should be fixed by `f145e7c`

File: infer/lib/audio.py (hunk 2 above)
Comment (413x1nkp) [reply infer/lib/audio.py]: Should be fixed by `f145e7c`

File: infer/lib/audio.py [hunk 3]
```diff
@@ -16,9 +16,9 @@
 
 def wav2(i: BytesIO, o: BufferedWriter, format: str):
     inp = av.open(i, "r")
-    format = video_format_dict[format]
+    format = video_format_dict.get(format) or "mp4"
```
Comment (fumiama) [inline infer/lib/audio.py]: Well, just `get(format, format)`, if my memory is correct.

**Review by fumiama — CHANGES_REQUESTED** [review]

File: infer/lib/audio.py (hunk 3 above)
Comment (413x1nkp) [reply infer/lib/audio.py]: Huh, I didn't know Python supported that
Will change.

File: infer/lib/audio.py [hunk 4]
```diff
@@ -1,18 +1,24 @@
...
+audio_format_dict: Dict[str, str] = {
+    "ogg": "libvorbis",
+    "mp4": "aac",
+}
+
+def wav2(i: BytesIO, o: BufferedWriter, format: str):
     inp = av.open(i, "r")
-    if format == "m4a":
-        format = "mp4"
+    format = video_format_dict.get(format, "mp4")
```
Comment (fumiama) [inline infer/lib/audio.py]: Well, well, just `get(format, format)` 😂. You can run a test on this grammar.

**Review by fumiama — CHANGES_REQUESTED** [review]

File: infer/lib/audio.py (hunk 4 above)
Comment (413x1nkp) [reply infer/lib/audio.py]: oh... OHHHHHH
Sorry :sweat_smile:

**Review by fumiama — APPROVED** [review]
You can see that this file used two ffmepg impls, one is av, the other is ffmpeg. I would like to replace all the ffmpeg usage (not so much, just one or two) to av so that We can drop the need to download ffmpeg at the root dir or path. If you are also interesting about this, you can have a try.

**Comment by 413x1nkp** [comment]
> You can see that this file used two ffmepg impls, one is av, the other is ffmpeg. I would like to replace all the ffmpeg usage (not so much, just one or two) to av so that We can drop the need to download ffmpeg at the root dir or path. If you are also interesting about this, you can have a try.

That's a great idea, to be honest! I'll try to do that, but a lot of the audio-related operations are all so... decentralized right now
But I'm willing to give it a try! :heart:

**Comment by fumiama** [comment]
> > You can see that this file used two ffmepg impls, one is av, the other is ffmpeg. I would like to replace all the ffmpeg usage (not so much, just one or two) to av so that We can drop the need to download ffmpeg at the root dir or path. If you are also interesting about this, you can have a try.
> 
> 
> 
> That's a great idea, to be honest! I'll try to do that, but a lot of the audio-related operations are all so... decentralized right now
> 
> But I'm willing to give it a try! :heart: 

You can use the search function in your IDE and just search the string `ffmpeg.` (donnot forget the last dot). The result is all the usage. One is just in this file, the other is a call to `ffprobe`.

**Comment by 413x1nkp** [comment]
> You can use the search function in your IDE and just search the string `ffmpeg.` (donnot forget the last dot). The result is all the usage. One is just in this file, the other is a call to `ffprobe`.

Created a PR introducing this change. #31

**Pull request merged**